internal/provider/statuspage.go

```diff
@@ -110,6 +111,67 @@ func isHCPOperational() (diags diag.Diagnostics) {
 	return diags
 }
 
+func isHCPOperationalFramework() (diags diagnostic.Diagnostics) {
```

Comment:
I ended up needing to effectively duplicate this function but return the proper diagnostics from the plugin framework package, so this could be used in the configure in `provider_framework.go`

Reply:
Makes sense. Mind leaving a comment?

Reply:
Absolutely!